documentation/modules/appendix/pages/model-evolution.adoc

```diff
@@ -0,0 +1,232 @@
+thus excluded from the Aspect Model, or that should have a different structure,
+. In the original model, something was modelled in the wrong way,
+. As part of the natural evolution of the model when the initial minimal subset of Properties is
+expanded to a comprehensive description of the domain,
+. etc.
+
+It is to be expected that an Aspect Model is continually improved upon (to a certain point, when it
+is considered "finished"). The model maintainers can change the Aspect Model as they see fit and
+release a new version. But what happens with the running Aspects and clients? Do all of them have to
+be changed immediately every time?
+
+[[decoupling-of-models-and-payload]]
+== Decoupling of Models and Payload
+
+The first measure towards a graceful evolution of Aspect Models in a living system (i.e., the models
+don't live in a vacuum but are actually used by Aspects and Clients) is the decoupling of models and
+payload. In the section that describes the xref:ROOT:payloads#mapping-to-json[mapping of Aspect
+models to corresponding JSON payloads], you can see that the JSON payloads have no indication of the
+Aspect Model that describes them. In particular, they do not contain a reference to the URN of the
+corresponding Aspect Model. This is similar to how the structure of a JSON document can be validated
+against a JSON Schema.footnote:[However, a JSON Schema only describes a document's structure, not
+its meaning or content.]
+
+Therefore there are changes that can be made in the Aspect Model so that the structure of the
+corresponding JSON payload does not change. We will call this type of changes _safe changes_. When a
+new version of an Aspect Model is released that contains only safe changes and a corresponding
+Aspect delivers only data that is valid in regards to the new version of the Aspect Model, the data
+is also still valid in regards to the previous version of the Aspect Model. The Aspect can therefore
+still be used by clients built using the previous version _without having to be changed_.
+
+At the place used to look up the Aspect, the previous version of the Aspect Model and the updated
+version can and should both be associated simultaneously with the same Aspect endpoint. This allows
+clients that only understand the previous version of the Aspect Model to still consume data from the
+Aspect (although they do not benefit from the updates introduced in the new model version), while
+clients that are updated to use the new Aspect Model can benefit from the update.
+
+At some point, older versions of the model can be removed from the Aspect. However, the exact
+conditions of when this can be done depend heavily on the concrete use case and are therefore out of
+scope of this document.
+
+[[safe-changes]]
+== Safe Changes for Graceful Model Evolution
+
+Every change to the Aspect Model that does not change the JSON payload corresponding to the model is
```

Comment:
"does not change" is not a correct statement for mandatory to optional.

Something like: "that permits all JSON payload previously correct as still correct .

Reply:
... "that is backwards compatible in the sense that it permits all previously permitted JSON payload AND keeps semantics intact (see below)"